src/Error/UnexpectedResponse.php

```diff
@@ -0,0 +1,39 @@
+<?php declare(strict_types=1);
+
+namespace Spawnia\Sailor\Error;
+
+use Psr\Http\Message\ResponseInterface;
+
+class UnexpectedResponse extends \Exception
+{
+    public int $statusCode = 0;
+
+    public string $responseBody = '';
+
+    /** @var array<string, list<string>> */
+    public array $responseHeaders;
+
+    public static function statusCode(
```

Comment:
Have you planned other methods, and what would those be called?

Reply:
e.g.

`public static function bodyContentsFormat(string $bodyContents): self`

`public static function clientException(ClientExceptionInterface $clientException): self`